test/systematic.jl

```diff
@@ -0,0 +1,76 @@
+using ChainRulesCore
+using ChainRulesTestUtils
+using ForwardDiff
+using ImplicitDifferentiation
+using JET
+using LinearAlgebra
+using Random
+using StaticArrays
+using Test
+using Zygote
+
+Random.seed!(63);
+
+"""
+    mysqrt(x)
+
+Compute the elementwise square root, breaking Zygote.jl and ForwardDiff.jl in the process.
+"""
+function mysqrt(x::AbstractArray)
+    a = [0.0]
+    a[1] = first(x)
+    return sqrt.(x)
+end
+
+function make_implicit_sqrt(linear_solver)
+    forward(x) = mysqrt(x)
+    conditions(x, y) = y .^ 2 .- x
+    implicit = ImplicitFunction(forward, conditions, linear_solver)
+    return implicit
+end
+
+function test_implicit_sqrt_call(implicit, x)
+    @test (@inferred implicit(x)) ≈ sqrt.(x)
+    if VERSION >= v"1.9"
+        test_opt(implicit, (typeof(x),))
+    end
+end
+
+function test_implicit_sqrt_forward(implicit, x)
+    y = implicit(x)
+    J = Diagonal(0.5 ./ vec(sqrt.(x)))
+    @test ForwardDiff.jacobian(implicit, x) ≈ J
+    x_and_dx = ForwardDiff.Dual.(x, ((0, 0),))
+    res_and_dres = @inferred implicit(x_and_dx)
```

Comment:
we should check that this is static when the input is static

Reply:
done